### PR TITLE
feat: generate mobile friendly config

### PR DESCRIPTION
**Pull request by sobolk**
**Issue #, if available:**

https://app.asana.com/0/1205484060153101/1205931466811581/f

**Description of changes:**

This PR 
1. adds initial version of iOS/Android config by transforming JS config into desired schema
2. adds `json-mobile` format
3. fixes dart format to conform to the same format as other mobile platform

**Assumptions**

1. This is a solution until frontend format is unified in https://app.asana.com/0/1205484060153101/1205931466811588/f 
2. Data parity between JS and mobile format can be achieved by adding necessary fields to JS config and transforming them into mobile.

**Out of scope**

This PR does not attempt to unify format across JS and mobile libraries. Neither it tries to change "config contributors" to use "generalized format" and keeps using JS as externally facing (what generateConfig returns) format.

**Testing**

1. Added unit test
4. Manual sanity check with android app.
    1. Created android app using gen1 tutorials
    5. Replaced amplify configuration using Samsara generated config

The android app has been created using these tutorials:
https://docs.amplify.aws/lib/auth/getting-started/q/platform/android/
https://docs.amplify.aws/lib/graphqlapi/query-data/q/platform/android/#query-item

data.ts
```
import { type ClientSchema, a, defineData } from '@aws-amplify/backend';

const schema = `
enum Priority {
  LOW
  NORMAL
  HIGH
}

type Todo @model
@auth(rules: [{ allow: public}]) {
  id: ID!
  name: String!
  priority: Priority!
  createdAt: AWSDateTime!
  completedAt: AWSDateTime
}
`
//
// export type Schema = ClientSchema<typeof schema>;

export const data = defineData({ schema,   authorizationModes: {
        defaultAuthorizationMode: "API_KEY",
        apiKeyAuthorizationMode: {
            expiresInDays: 7
        }
    }  });

```

auth.ts
```
import { defineAuth } from '@aws-amplify/backend';

export const auth = defineAuth({
  loginWith: {
    email: true,
  },
});

```

MyAmplifyApp.kt
```
package com.example.todo

import android.app.Application
import android.util.Log
import com.amplifyframework.AmplifyException
import com.amplifyframework.core.Amplify
import com.amplifyframework.auth.cognito.AWSCognitoAuthPlugin
import com.amplifyframework.api.aws.AWSApiPlugin

class MyAmplifyApp: Application() {

    override fun onCreate() {
        super.onCreate()

        try {
            Amplify.addPlugin(AWSCognitoAuthPlugin())
            Amplify.addPlugin(AWSApiPlugin())
            Amplify.configure(applicationContext)

            Log.i("Tutorial", "Initialized Amplify")
        } catch (error: AmplifyException) {
            Log.e("Tutorial", "Could not initialize Amplify", error)
        }
    }
}
```

MainActivity.kt
```
package com.example.todo

import android.util.Log
import com.amplifyframework.core.Amplify
import com.amplifyframework.core.model.temporal.Temporal
import com.amplifyframework.datastore.generated.model.Priority
import com.amplifyframework.datastore.generated.model.Todo
import java.util.*
import java.util.concurrent.TimeUnit

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.amplifyframework.api.graphql.model.ModelMutation
import com.amplifyframework.api.graphql.model.ModelQuery
import com.amplifyframework.auth.cognito.result.AWSCognitoAuthSignOutResult
import com.amplifyframework.core.model.query.Where
import com.example.todo.ui.theme.TodoTheme

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            TodoTheme {
                // A surface container using the 'background' color from the theme
                Surface(modifier = Modifier.fillMaxSize(), color = MaterialTheme.colorScheme.background) {
                    Greeting("Android")
                }
            }
        }

        Log.i("AuthQuickstart", "================================")
        Log.i("AuthQuickstart", "================================")
        Log.i("AuthQuickstart", "================================")
        Log.i("AuthQuickstart", "================================")



//        val options = AuthSignUpOptions.builder()
//            .userAttribute(AuthUserAttributeKey.email(), "REDACTED@amazon.com")
//            .build()
//        Amplify.Auth.signUp("REDACTED@amazon.com", "REDACTED", options,
//            { Log.i("AuthQuickStart", "Sign up succeeded: $it") },
//            { Log.e ("AuthQuickStart", "Sign up failed", it) }
//        )

//        Amplify.Auth.confirmSignUp(
//            "REDACTED@amazon.com", "REDACTED",
//            { result ->
//                if (result.isSignUpComplete) {
//                    Log.i("AuthQuickstart", "Confirm signUp succeeded")
//                } else {
//                    Log.i("AuthQuickstart","Confirm sign up not complete")
//                }
//            },
//            { Log.e("AuthQuickstart", "Failed to confirm sign up", it) }
//        )

        Amplify.Auth.signOut { signOutResult ->
            when(signOutResult) {
                is AWSCognitoAuthSignOutResult.CompleteSignOut -> {
                    // Sign Out completed fully and without errors.
                    Log.i("AuthQuickStart", "Signed out successfully")
                }
                is AWSCognitoAuthSignOutResult.PartialSignOut -> {
                    // Sign Out completed with some errors. User is signed out of the device.
                    signOutResult.hostedUIError?.let {
                        Log.e("AuthQuickStart", "HostedUI Error", it.exception)
                        // Optional: Re-launch it.url in a Custom tab to clear Cognito web session.

                    }
                    signOutResult.globalSignOutError?.let {
                        Log.e("AuthQuickStart", "GlobalSignOut Error", it.exception)
                        // Optional: Use escape hatch to retry revocation of it.accessToken.
                    }
                    signOutResult.revokeTokenError?.let {
                        Log.e("AuthQuickStart", "RevokeToken Error", it.exception)
                        // Optional: Use escape hatch to retry revocation of it.refreshToken.
                    }
                }
                is AWSCognitoAuthSignOutResult.FailedSignOut -> {
                    // Sign Out failed with an exception, leaving the user signed in.
                    Log.e("AuthQuickStart", "Sign out Failed", signOutResult.exception)
                }
            }
        }

        Amplify.Auth.signIn("REDACTED@amazon.com", "REDACTED",
            { result ->
                if (result.isSignedIn) {
                    Log.i("AuthQuickstart", "Sign in succeeded")
                } else {
                    Log.i("AuthQuickstart", "Sign in not complete")
                }
            },
            { Log.e("AuthQuickstart", "Failed to sign in", it) }
        )

        Amplify.Auth.fetchAuthSession(
            { Log.i("AmplifyQuickstart", "Auth session = $it") },
            { error -> Log.e("AmplifyQuickstart", "Failed to fetch auth session", error) }
        )

        val todo = Todo.builder()
            .name("My todo " + UUID.randomUUID().toString())
            .priority(Priority.HIGH)
            .build()

        Amplify.API.query(
            ModelQuery.list(Todo::class.java),
            { response ->
                response.data.forEach { todo ->
                    Log.i("MyAmplifyApp", todo.name)
                }
            },
            { Log.e("MyAmplifyApp", "Query failure", it) }
        )

        Amplify.API.mutate(
            ModelMutation.create(todo),
            { Log.i("MyAmplifyApp", "Added Todo with id: ${it.data.id}") },
            { Log.e("MyAmplifyApp", "Create failed", it) }
        )
    }
}

@Composable
fun Greeting(name: String, modifier: Modifier = Modifier) {
    Text(
            text = "Hello $name!",
            modifier = modifier
    )
}

@Preview(showBackground = true)
@Composable
fun GreetingPreview() {
    TodoTheme {
        Greeting("Android")
    }
}
```

Log:
```
023-11-14 10:11:05.626  9463-9463  ziparchive              com.example.todo                     W  Unable to open '/data/data/com.example.todo/code_cache/.overlay/base.apk/classes3.dm': No such file or directory
2023-11-14 10:11:05.627  9463-9463  ziparchive              com.example.todo                     W  Unable to open '/data/app/~~6TbJyN45kfJlJ1PvTTe5Jg==/com.example.todo-enHcEHt0ElKqYNGxv1drDw==/base.dm': No such file or directory
2023-11-14 10:11:05.627  9463-9463  ziparchive              com.example.todo                     W  Unable to open '/data/app/~~6TbJyN45kfJlJ1PvTTe5Jg==/com.example.todo-enHcEHt0ElKqYNGxv1drDw==/base.dm': No such file or directory
2023-11-14 10:11:05.956  9463-9463  nativeloader            com.example.todo                     D  Configuring clns-6 for other apk /data/app/~~6TbJyN45kfJlJ1PvTTe5Jg==/com.example.todo-enHcEHt0ElKqYNGxv1drDw==/base.apk. target_sdk_version=33, uses_libraries=, library_path=/data/app/~~6TbJyN45kfJlJ1PvTTe5Jg==/com.example.todo-enHcEHt0ElKqYNGxv1drDw==/lib/arm64, permitted_path=/data:/mnt/expand:/data/user/0/com.example.todo
2023-11-14 10:11:05.969  9463-9463  GraphicsEnvironment     com.example.todo                     V  Currently set values for:
2023-11-14 10:11:05.969  9463-9463  GraphicsEnvironment     com.example.todo                     V    angle_gl_driver_selection_pkgs=[]
2023-11-14 10:11:05.970  9463-9463  GraphicsEnvironment     com.example.todo                     V    angle_gl_driver_selection_values=[]
2023-11-14 10:11:05.970  9463-9463  GraphicsEnvironment     com.example.todo                     V  ANGLE GameManagerService for com.example.todo: false
2023-11-14 10:11:05.970  9463-9463  GraphicsEnvironment     com.example.todo                     V  com.example.todo is not listed in per-application setting
2023-11-14 10:11:05.971  9463-9463  GraphicsEnvironment     com.example.todo                     V  Neither updatable production driver nor prerelease driver is supported.
2023-11-14 10:11:06.116  9463-9463  Tutorial                com.example.todo                     I  Initialized Amplify
2023-11-14 10:11:06.141  9463-9489  libEGL                  com.example.todo                     D  loaded /vendor/lib64/egl/libEGL_emulation.so
2023-11-14 10:11:06.171  9463-9489  libEGL                  com.example.todo                     D  loaded /vendor/lib64/egl/libGLESv1_CM_emulation.so
2023-11-14 10:11:06.173  9463-9489  libEGL                  com.example.todo                     D  loaded /vendor/lib64/egl/libGLESv2_emulation.so
2023-11-14 10:11:06.175  9463-9463  AuthQuickstart          com.example.todo                     I  ================================
2023-11-14 10:11:06.175  9463-9463  AuthQuickstart          com.example.todo                     I  ================================
2023-11-14 10:11:06.175  9463-9463  AuthQuickstart          com.example.todo                     I  ================================
2023-11-14 10:11:06.176  9463-9463  AuthQuickstart          com.example.todo                     I  ================================
2023-11-14 10:11:06.268  9463-9482  EngineFactory           com.example.todo                     I  Provider GmsCore_OpenSSL not available
2023-11-14 10:11:06.323  9463-9463  Compatibil...geReporter com.example.todo                     D  Compat change id reported: 237531167; UID 10190; state: DISABLED
2023-11-14 10:11:06.327  9463-9463  OpenGLRenderer          com.example.todo                     W  Unknown dataspace 0
2023-11-14 10:11:06.394  9463-9463  om.example.todo         com.example.todo                     W  Method java.lang.Object androidx.compose.runtime.snapshots.SnapshotStateMap.mutate(kotlin.jvm.functions.Function1) failed lock verification and will run slower.
                                                                                                    Common causes for lock verification issues are non-optimized dex code
                                                                                                    and incorrect proguard optimizations.
2023-11-14 10:11:06.395  9463-9463  om.example.todo         com.example.todo                     W  Method void androidx.compose.runtime.snapshots.SnapshotStateMap.update(kotlin.jvm.functions.Function1) failed lock verification and will run slower.
2023-11-14 10:11:06.395  9463-9463  om.example.todo         com.example.todo                     W  Method boolean androidx.compose.runtime.snapshots.SnapshotStateMap.removeIf$runtime_release(kotlin.jvm.functions.Function1) failed lock verification and will run slower.
2023-11-14 10:11:06.474  9463-9483  System.err              com.example.todo                     W  SLF4J: No SLF4J providers were found.
2023-11-14 10:11:06.487  9463-9483  System.err              com.example.todo                     W  SLF4J: Defaulting to no-operation (NOP) logger implementation
2023-11-14 10:11:06.488  9463-9483  System.err              com.example.todo                     W  SLF4J: See https://www.slf4j.org/codes.html#noProviders for further details.
2023-11-14 10:11:06.518  9463-9487  OpenGLRenderer          com.example.todo                     W  Failed to choose config with EGL_SWAP_BEHAVIOR_PRESERVED, retrying without...
2023-11-14 10:11:06.518  9463-9487  OpenGLRenderer          com.example.todo                     W  Failed to initialize 101010-2 format, error = EGL_SUCCESS
2023-11-14 10:11:06.527  9463-9487  Gralloc4                com.example.todo                     I  mapper 4.x is not supported
2023-11-14 10:11:06.534  9463-9487  OpenGLRenderer          com.example.todo                     E  Unable to match the desired swap behavior.
2023-11-14 10:11:06.640  9463-9513  TrafficStats            com.example.todo                     D  tagSocket(91) with statsTag=0xffffffff, statsUid=-1
2023-11-14 10:11:06.731  9463-9513  TrafficStats            com.example.todo                     D  tagSocket(93) with statsTag=0xffffffff, statsUid=-1
2023-11-14 10:11:06.877  9463-9518  TrafficStats            com.example.todo                     D  tagSocket(109) with statsTag=0xffffffff, statsUid=-1
2023-11-14 10:11:06.895  9463-9513  TrafficStats            com.example.todo                     D  tagSocket(109) with statsTag=0xffffffff, statsUid=-1
2023-11-14 10:11:07.266  9463-9500  MyAmplifyApp            com.example.todo                     I  Added Todo with id: ae086b7e-2488-41c6-87e9-fe88e35d784e
2023-11-14 10:11:07.269  9463-9483  AuthQuickStart          com.example.todo                     I  Signed out successfully
2023-11-14 10:11:07.328  9463-9499  MyAmplifyApp            com.example.todo                     I  My todo de2866d2-7823-48d3-a351-9fc4179f7132
2023-11-14 10:11:07.328  9463-9499  MyAmplifyApp            com.example.todo                     I  My first todo
2023-11-14 10:11:07.328  9463-9499  MyAmplifyApp            com.example.todo                     I  My first todo
2023-11-14 10:11:07.328  9463-9499  MyAmplifyApp            com.example.todo                     I  My first todo
2023-11-14 10:11:07.329  9463-9499  MyAmplifyApp            com.example.todo                     I  My todo 6d42618f-5728-48dd-9507-84b65bda09dc
2023-11-14 10:11:07.700  9463-9524  TrafficStats            com.example.todo                     D  tagSocket(115) with statsTag=0xffffffff, statsUid=-1
2023-11-14 10:11:07.706  9463-9528  TrafficStats            com.example.todo                     D  tagSocket(115) with statsTag=0xffffffff, statsUid=-1
2023-11-14 10:11:07.812  9463-9532  AuthQuickstart          com.example.todo                     I  Sign in succeeded
2023-11-14 10:11:07.820  9463-9482  AmplifyQuickstart       com.example.todo                     I  Auth session = AWSCognitoAuthSession(isSignedIn=true, identityIdResult=AuthSessionResult{value=null, error=ConfigurationException{message=Could not retrieve Identity ID, cause=null, recoverySuggestion=Cognito Identity not configured. Please check amplifyconfiguration.json file.}, type=FAILURE}, awsCredentialsResult=AuthSessionResult{value=null, error=ConfigurationException{message=Could not fetch AWS Cognito credentials, cause=null, recoverySuggestion=Cognito Identity not configured. Please check amplifyconfiguration.json file.}, type=FAILURE}, userSubResult=AuthSessionResult{value=f18f8906-937b-4ab8-b9ca-206cfd0a12c7, error=null, type=SUCCESS}, userPoolTokensResult=AuthSessionResult{value=AWSCognitoUserPoolTokens(accessToken=eyJraWQiOiJRMldBR0JDQmtwbFwvbDJabko1R29JMkZGTHFVTHAzYUZIRDE3M29uTW84OD0iLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiJmMThmODkwNi05MzdiLTRhYjgtYjljYS0yMDZjZmQwYTEyYzciLCJpc3MiOiJodHRwczpcL1wvY29nbml0by1pZHAudXMtd2VzdC0yLmFtYXpvbmF3cy5jb21cL3VzLXdlc3QtMl9kb2dMZ3VzNFIiLCJjbGllbnRfaWQiOiJqc29sb2U5ZGZ2aTNtdThmc3BjcWUyYzRxIiwib3JpZ2luX2p0aSI6ImZhYjA2NDdlLWIwZTgtNGM3Zi04NWU0LTE4YWQ1OGU2YjhlYSIsImV2ZW50X2lkIjoiNjBiM2Q5NzEtNDRlNi00ZDc1LTk3Y2EtNTY0MGMxZWY0ZmYxIiwidG9rZW5fdXNlIjoiYWNjZXNzIiwic2NvcGUiOiJhd3MuY29nbml0by5zaWduaW4udXNlci5hZG1pbiIsImF1dGhfdGltZSI6MTY5OTk4NTQ2NywiZXhwIjoxNjk5OTg5MDY3LCJpYXQiOjE2OTk5ODU0NjgsImp0aSI6ImE4NDYwMTU5LWRjMDMtNGFiNi1iMzhhLWYzN2VlOWFjMTg4ZCIsInVzZXJuYW1lIjoiZjE4Zjg5MDYtOTM3Yi00YWI4LWI5Y2EtMjA2Y2ZkMGExMmM3In0.p6gdINAbz_9y65rLYCNkJ_eqIeXQsVd100xPI1b2H4g8VrqNlRlC24fcXol3pgQiFz6F3KKiUtB4qWAv_df_uA8oSjb0rR7sljPOMIBl7zl967f7_R_g_ZO_3DZwGakNmy7r4ma_Smo8ZdhXos_WOk2lAtE_NzxQZjshc1BKDaLmHXv-qJUPVJfX6YZVyCA9mutXUnD4bysnHwTpzF1ZqY6Sw_un_xY_CzDiEdpCFi9wnhdet1xkPeq4jxW60KWwV2cyoCEOEdyQhkj_kMhkdb0JvTFZp0wFUzza2jDBQB4h3nEVtP8fuJfp4Dd2RMIu62Fb7YZjNq0A9TgQOTnfXw, idToken=eyJraWQiOiJCOFYzYlhrYmcxWGtYRlwvYTB6VXFxQlJRRysxZzZwdVU3bytNRDdjRGE4OD0iLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiJmMThmODkwNi05MzdiLTRhYjgtYjljYS0yMDZjZmQwYTEyYzciLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiaXNzIjoiaHR0cHM6XC9cL2NvZ25pdG8taWRwLnVzLXdlc3QtMi5hbWF6b25hd3MuY29tXC91cy13ZXN0LTJfZG9nTGd1czRSIiwiY29nbml0bzp1c2VybmFtZSI6ImYxOGY4OTA2LTkzN2ItNGFiOC1iOWNhLTIwNmNmZDBhMTJjNyIsIm9yaWdpbl9qdGkiOiJmYWIwNjQ3ZS1iMGU4LTRjN2YtODVlNC0xOGFkNThlNmI4ZWEiLCJhdWQiOiJqc29sb2U5ZGZ2aTNtdThmc3BjcWUyYzRxIiwiZXZlbnRfaWQiOiI2MGIzZDk3MS00NGU2LTRkNzUtOTdjYS01NjQwYzFlZjRmZjEiLCJ0b2tlbl91c2UiOiJpZCIsImF1dGhfdGltZSI6MTY5OTk4NTQ2NywiZXhwIjoxNjk5OTg5MDY3LCJpYXQiOjE2OTk5ODU0NjgsImp0aSI6IjdmZjI1YjBhLWZjMjktNDZlNy05N2FmLWM0OGJhZTc4ZGQ1OSIsImVtYWlsIjoic29ia2FtaWwrYW5kcm9pZEBhbWF6b24uY29tIn0.nF6JAD5d_1Kmp1XC400yvloO2PjhWy8IFLjOLHQ5PeVA7eW04JJ428t3-iuuwReETeGohuHGrVl1ZY6gxKlJTE0tKwaSSDxhRr8dws_xid5iVjXkEAEQY3HcE7Y8VIPqWTBHkCPS5kQiUFrl6DTcxUPuaKyHVuR9emuH8CXmco6B_WCsDRk8R_sbSnhNPp3fwmaPsNcscgdNWP_bwyMfHSi8tAy-rC2w3ZIrGjtPnPmx-t_zFldHiU5Uze9jSGx02XdZpZ3pibTYSLHQ3dC6xMt8E-GIFHC_N3BWMwgwsDbKmJsHxylB-cVhHjrJE7Jw895z2HGwf8PmEh4g9PWHgQ, refreshToken=eyJjdHkiOiJKV1QiLCJlbmMiOiJBMjU2R0NNIiwiYWxnIjoiUlNBLU9BRVAifQ.JJcjoADpwfrcZuoKoevuhKSD2rHUXrB3MCUrdKGUlqxgqSHYraBLT_d5M2lZJtQopAfHazXOJ2GQpmDeQfsCmu2XE9CevTU-EPu32sNCQHmHb_0eOsTczgtdBSvzFtzYwX8dIi23Q_DXKIuvPHF6cvzqafISjz5VpQ7AjX27QKY_piqflDUDrNc0nFR6eCpik13Cp34ti5lN4ZRikXv3vzjNmLftKETbO4pN5B844Ub_YbVS0NS92p0SaZbiLwIdoRsZtPRYP48_auzxkxtBA4LqxTuOeKFLzMvfa3j4oCxcxNCwLZzY-SWGRNw9GtdATvmJbW1iO9bs9VvCGCvlXw.LDCS7e3VrBKiE0MI.311goQ6RQuFgmDGCpJWQ96stwCqmyzUlFjaDqJmum_68OVao8LyFYazr-6huUhaR8vkMafJNG3p-UMVQJIO-HBzRfcQ6_hB5FlX1e9ea1y6kqfrgMH1oRbeZTC6HNsmUnyQEkZAAStmOC2lmydyHchHNXCTGC0JdgIoBC2LoFz4MRiJ6_rNb5vxp2qeFH2vaKWzdQaH5dmBqX_iiFgI7JPx9XqkEj0djwWj9OgdCCLCIlBa9ILuu24IgDXVGpN-IAiBIHm_oTPaMAd_QFdDvnzSqHzaleoFEz89ecdK6BjJ10Z85pSlJmROofCuzaG8Q1QceGcZnAnRJbtc7-Ym5EF25w8p4ui_VWn-1efc75VNlRQMpCp9oZrZzD8mL1ZZoKVy9rcLuYv1mY9X14lo9S1MGdC1Nu9fsLkhqn9UDLA5Y3aT7OK3YpoDHI7OKZFKB9dzN9HCv8cZT4_BAtmuMqqOF4GA5zgLc5vBC8VkHY-bSdSKluaAvbjmUQWHUccmhLZstWlHRYbKTl6ixzwWezyNJ447lchg96TpQrSlWRMm6LGs5RZ5ChGlQledWfupRP2gE8UcbP_xPWF_jaY1-P2zJD9Oa2sxdAg_uf45k9GClKwS1tzOUeO4DRoVTCz118N4hTcx10TwJrTJEBD-K31VeFf3UmTWpe5sjiBVQmFc-ZI1fgp6fGXQreX
2023-11-14 10:11:11.637  9463-9537  ProfileInstaller        com.example.todo                     D  Installing profile for com.example.todo
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/client-config/API.md [hunk 1]
```diff
@@ -26,6 +26,8 @@ export enum ClientConfigFormat {
     // (undocumented)
     JSON = "json",
     // (undocumented)
+    JSON_MOBILE = "json-mobile",
```
Comment (sobolk) [inline packages/client-config/API.md:29]: https://app.asana.com/0/1205484060153101/1205931466811581/f

The other option was to add additional dimension like `--target` but that was rejected.

This should go away if/when client config is unified here https://app.asana.com/0/1205484060153101/1205931466811588/f

File: packages/client-config/src/generate_client_config_to_file.ts [hunk 2]
```diff
@@ -14,12 +18,28 @@ export const generateClientConfigToFile = async (
...
   const clientConfig = await generateClientConfig(
     credentialProvider,
     backendIdentifier
   );
-  const targetPath = await getClientConfigPath(outDir, format);
-  await clientConfigWriter.writeClientConfig(clientConfig, targetPath);
+  await clientConfigWriter.writeClientConfig(clientConfig, outDir, format);
+};
+
+const readPackageJson = async (): Promise<{
```
Comment (edwardfoyle) [inline packages/client-config/src/generate_client_config_to_file.ts:37]: we have `loadCwdPackageJson` in the cli package. Not sure if this duplication warrants pulling that into `plaform-core` but something to consider

**Review by edwardfoyle — APPROVED** [review]

File: packages/client-config/src/generate_client_config_to_file.ts (hunk 2 above)
Comment (Amplifiyer) [reply packages/client-config/src/generate_client_config_to_file.ts:37]: I'm doing that as part of my usage metrics PR, will pull it in once this is merged.

**Review by Amplifiyer — APPROVED** [review]

File: packages/client-config/src/generate_client_config_to_file.ts (hunk 2 above)
Comment (sobolk) [reply packages/client-config/src/generate_client_config_to_file.ts:37]: I looked at `loadCwdPackageJson` but what we need here is really `loadMyPackageJson` not cwd, which is first occurence of this (if we set aside typing of package json).

@Amplifiyer thanks!

**Pull request merged**